onecodex/viz/_metadata.py

```diff
@@ -1,83 +1,319 @@
+            - 'kingdom' or others: restrict analysis to taxa at this rank
+        normalize (bool): convert from read counts to relative abundances (each sample sums to 1.0)
+
+    Options for plotting:
+        label (string) -- metadata field to label samples with
+        title (string) -- main title of the plot
+        xlabel, ylabel (string) -- axes labels
+        boxplot (bool) -- force output a box plot
+        scatter (bool) -- force output a scatter plot
     """
-    if not tax_id and not statistic:
-        raise OneCodexException('Please pass a `tax_id` or a `statistic`.')
-    elif tax_id and statistic:
-        raise OneCodexException('Please pass only a `tax_id` or a `statistic`.')
-
-    sns.set(style=kwargs.pop('style', 'darkgrid'))
-
-    normed_classifications, md = normalize_classifications(analyses, label=label)
-
-    if metadata not in md:
-        raise OneCodexException('Selected metadata field `{}` not available'.format(metadata))
-
-    stat = []
-    if statistic:
-        for analysis in normed_classifications:
-            try:
-                stat.append(alpha_diversity(analysis, statistic, field=field, rank=rank))
-            except ValueError:
-                raise NotImplementedError('{} statistic are currently supported.'.format(statistic))
-    elif tax_id:
-        df, tax_info = collate_classification_results(normed_classifications, field=field, rank=None,
-                                                      normalize=normalize)
-        stat = df.loc[:, str(tax_id)].values
-        if stat.shape[0] == 0:
-            raise OneCodexException('No values found for `tax_id` {} and `field` {}.'.format(tax_id, field))
-
-    md['_data'] = stat
-
-    # Try to plot any column with `date` in it as a datetime, or if it's already a datetime
-    # Plot numeric types as lmplots
-    # Plot categorical, boolean, and objects as boxplot
-    if 'date' in metadata.split('_') or pd.api.types.is_datetime64_any_dtype(md[metadata]):
-        if not pd.api.types.is_datetime64_any_dtype(md[metadata]):
-            md.loc[:, metadata] = md.loc[:, metadata].apply(pd.to_datetime, utc=True)
-        fig, ax = plt.subplots()
-        ax.xaxis.set_major_locator(mdates.AutoDateLocator())
-        ax.xaxis.set_major_formatter(mdates.DateFormatter('%m/%d/%Y'))
-        ax.plot_date(md[metadata].values, md['_data'].values, **kwargs)
-        fig.autofmt_xdate()
-    elif pd.api.types.is_numeric_dtype(md[metadata]):
-        sns.lmplot(x=metadata, y='_data', truncate=True, data=md, **kwargs)
-    elif pd.api.types.is_bool_dtype(md[metadata]) or \
-            pd.api.types.is_categorical_dtype(md[metadata]) or \
-            pd.api.types.is_object_dtype(md[metadata]):
-        na = {field: 'N/A' for field in md.columns}
-        md.fillna(value=na, inplace=True)
-        sns.boxplot(x=metadata, y='_data', data=md, palette='pastel', **kwargs)
+
+    if sum(map(bool, [statistic, metadata2, tax_id, tax_name])) in (0, 2, 3, 4):
```

Comment:
Cool! Maybe we should make it a set instead of a tuple though?

Reply:
... Can we just check `!= 1`? on this?

Reply:
Agreed, changed.